indico/web/forms/fields.py

```diff
@@ -238,16 +241,71 @@ def _serialize_principal(self, principal):
 
     def _value(self):
         if self.serializable:
-            data = retrieve_principals(self._get_data(), legacy=False)
+            principals = retrieve_principals(self._get_data(), legacy=False)
         else:
-            data = self._get_data()
-        data.sort(key=lambda x: x.name.lower())
-        return map(self._serialize_principal, data)
+            principals = self._get_data()
+        principals.sort(key=lambda x: x.full_name.lower())
+        return map(self._serialize_principal, principals)
 
     def _get_data(self):
         return sorted(self.data) if self.data else []
 
 
+class EventPersonListField(PrincipalListField):
+    """"A field that lets you select a list Indico user and EventPersons
+
+    Requires its form to have an event set.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(EventPersonListField, self).__init__(*args,
+                                                   groups=False, allow_external=True, serializable=False, **kwargs)
+
+    @property
+    def event(self):
+        return self.get_form().event
+
+    def _convert_data(self, data):
+        return map(self._get_event_person, data)
+
+    def _create_event_person(self, data):
+        user = User.find_first(~User.is_deleted, User.all_emails.contains(data['email']))
+        title = next((x.value for x in UserTitle if data.get('title') == x.title), None)
+        return EventPerson(event_new=self.event, user=user, email=data['email'], _title=title,
+                           first_name=data['firstName'], last_name=data['familyName'],
+                           affiliation=data.get('affiliation'), address=data.get('address'),
+                           phone=data.get('phone'))
+
+    def _get_event_person(self, data):
+        person_type = data.get('_type')
+        if person_type is None:
+            return self._create_event_person(data)
+        if person_type == 'Avatar':
```

Comment:
To make it clear that you'll only get into one of those branches, even without looking at the `returss